Sources/PanelTransition/Driver/TransitionDriver.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  Copyright (c) 2020 Touch Instinct
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the Software), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+
+open class TransitionDriver: UIPercentDrivenInteractiveTransition {
+    private weak var presentedController: UIViewController?
+    
+    private var panRecognizer: UIPanGestureRecognizer?
+    
+    public var direction: TransitionDirection = .present
+    
+    // MARK: - Linking
+    public func link(to controller: UIViewController) {
+        presentedController = controller
+        
+        panRecognizer = UIPanGestureRecognizer(target: self, action: #selector(handleGesture))
+        presentedController?.view.addGestureRecognizer(panRecognizer!)
+    }
+    
+    // MARK: - Override
+    override open var wantsInteractiveStart: Bool {
+        get {
+            switch direction {
+            case .present:
+                return false
+            case .dismiss:
+                return panRecognizer?.state == .began
+            }
+        }
+        
+        set {}
+    }
+    
+    @objc private func handleGesture(recognizer: UIPanGestureRecognizer) {
+        switch direction {
+        case .present:
+            handlePresentation(recognizer: recognizer)
+        case .dismiss:
+            handleDismiss(recognizer: recognizer)
+        }
+    }
+}
+
+// MARK: - Gesture Handling
+private extension TransitionDriver {
+    var maxTranslation: CGFloat {
+        return presentedController?.view.frame.height ?? 0
+    }
+    
+    /// `pause()` before call `isRunning`
+    var isRunning: Bool {
+        return percentComplete != 0
```

Comment:
can add observer for `percentComplete` property?

Reply:
Is it needed?

Reply:
I think so, because there are cases where the logic depends of animation. 
you can override `percentComplete`, added `relay` and `observable`?
and yet add the enum state for this process with case open, close, process

Reply:
Определим, как фичу